### PR TITLE
Bug 1880246:Entry keys in aggregated Helm index are appended with the name of source Helm repo

### PR DESCRIPTION
**Pull request by pedjak**
The introduced convention enables that all charts with the same name, but
in different source Helm repo, appear in the aggregated index file under
the name following the pattern: `foo-chart--my-repo`, where `my-repo` is the name
of `HelmChartRepository` available in the cluster.

### CODE REVIEW

File: pkg/helm/chartproxy/proxy.go [hunk 1]
```diff
@@ -69,13 +69,15 @@ func (p *proxy) IndexFile() (*repo.IndexFile, error) {
...
-	var indexFiles []*repo.IndexFile
+	indexFile := repo.NewIndexFile()
 	for _, helmRepo := range helmRepos {
 		idxFile, err := helmRepo.IndexFile()
 		if err != nil {
 			return nil, err
 		}
-		indexFiles = append(indexFiles, idxFile)
+		for key, entry := range idxFile.Entries {
+			indexFile.Entries[key+"--"+helmRepo.Name] = entry
```
Comment (rohitkrai03) [inline pkg/helm/chartproxy/proxy.go:79]: Maybe we add '~~' as a delimiter between the two names to make it more unique?
```suggestion
			indexFile.Entries[key+"~~"+helmRepo.Name] = entry
```

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: pkg/helm/chartproxy/proxy.go (hunk 1 above)
Comment (pedjak) [reply pkg/helm/chartproxy/proxy.go:79]: this char is not alowed - only letters, digits and dash

Comment (rohitkrai03) [reply pkg/helm/chartproxy/proxy.go:79]: Okay, no problem.

**Review by rohitkrai03 — APPROVED** [review]
Looks good to me. Tested the backend changes with UI changes in https://github.com/openshift/console/pull/6761. Works well.

**Pull request merged**